### ISSUE TITLE
Internal Server Error when deleting a newly created Flow

### ISSUE THREAD

**Issue by gespinha** [directus/directus#14550]
### Describe the Bug

After installing a vanilla instance of Directus on MacOS 12.4, I created my first flow. Attempting to delete it, returns an error.

### To Reproduce

1. Install Directus via the terminal
2. Open the Settings page
3. Open the Flows page
4. Create a new flow
5. Add a couple of random nodes to the flow
6. Open the flow
7. Click "delete" button

Result: error prompt.

(this only happens when the flow I'm attempting to delete is the last one in the list)

### Errors Shown

`[INTERNAL_SERVER_ERROR] delete from `directus_flows` where `id` in ('4ef405a5-3255-4fc6-9a8b-194cddfe636e') - ER_ROW_IS_REFERENCED_2: Cannot delete or update a parent row: a foreign key constraint fails (`directus`.`directus_operations`, CONSTRAINT `directus_operations_reject_foreign` FOREIGN KEY (`reject`) REFERENCES `directus_operations` (`id`)) `

### What version of Directus are you using?

9.14.1

### What version of Node.js are you using?

14.15.1

### What database are you using?

MySQL 8.0.29

### What browser are you using?

Edge

### How are you deploying Directus?

running locally

**Comment by azrikahar** [comment]
Can you share a quick screenshot on how step 5 looks like in your flow? This does seem very much like to #14463.

Aside from that, can you try editing and removing operations inside that flow, either one by one until the flow deletion succeeds or all if that'll take too much time and attempt to delete the flow at the end?

**Comment by gespinha** [comment]
@azrikahar This is what I see when I'm editing the nodes:

![image](https://user-images.githubusercontent.com/4227944/180209378-1022c356-8581-40b1-8375-4658e24cd689.png)

No issues there. I can add, edit and remove nodes. My issue happens when I try to delete the entire flow, the first one I created (this issue only happens on the very first flow I created), which gives me this error:

![image](https://user-images.githubusercontent.com/4227944/180209587-c6dcd48a-6b11-4722-b7f0-ec423281d4ad.png)

**Comment by azrikahar** [comment]
@gespinha Thanks! Would you mind trying to delete the "Update Data" operation, save it, and try deleting the flow again?

**Comment by gespinha** [comment]
@azrikahar Yup, that worked! Just deleted the "Update Data" and saved the flow successfully. Thanks for the advice!

Is this a bug or was it something I did wrong?

**Comment by azrikahar** [comment]
@gespinha glad that worked out! I don't think you're doing anything wrong as I was able to reproduce this back in #14463, but I did had to make a second attempt to face this error. It seems specific to MySQL (and possibly MariaDB) at the moment, but we'll need to investigate further. Thanks again for confirmation that it's the (nested) operations causing this error on Flow deletion 👍
